pkgs/native_toolchain_c/test/clinker/objects_cross_test.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+//TODO(mosuem): Enable for windows and mac.
+// See https://github.com/dart-lang/native/issues/1376.
+@TestOn('linux')
```

Comment:
Should this be removed now?

Reply:
I believe this is still correct. This test checks whether we can link on the current host OS for different architectures running the same (host) OS. For that, we still only support Linux for now. The new functionality (cross linking for Android from Linux/Windows/MaxOS) I added the new *_cross_android_test.dart tests, which do not have a `@TestOn` limitation.

(Technically, this test doesn't belong to this change as it is unrelated to the Android work and testing existing functionality. I just saw this as a whole in the existing testing story and added it for completeness - and to make sure that my Android work doesn't break this scenario).